### DISCUSSION TITLE
Telemetry only RX - antenna tracker

### DISCUSSION THREAD

**Discussion by sdellava** [ExpressLRS/ExpressLRS#2340] [Ideas]
Good morning, everyone.

Airport is a great solution for receiving telemetry on a base station, however it requires a second transmitter and typically a PC on which the data can be viewed or used for various purposes. One possible purpose is to use the GPS position sent from the aircraft and pass it to an antenna tracker. The final system configuration is quite complex, and with this feature request I propose a solution that can reduce it significantly.

Airport transmits the data stream, produced by the FC, which contains the same information that all ELRS RXs can transmit.

Unfortunately, the limited transmitting power of the RXs makes telemetry impractical because with an omnidirectional antenna on the receiver and transmitter and 100 mW of transmitting power, after 400-500 m, the receiver can no longer decode the transmitted data. This amply justifies Airport, however, one could also replace the receiver antenna with a directive one. This idea requires precisely pointing the antenna toward the aircraft. Doing this manually is possible, but if you get distracted for a moment you risk failsaife.

We can overcome this problem by using a antenna tracker that can point a directional antenna toward the aircraft. Antennas for receiving video signals can be installed on the same tracker. 

The antenna trackers (such as Ardupilot's) however, use a protocol (Mavlink) that requires a special transmitter (basically requiring a link similar to Airport) 

We can overcome this problem by using a antenna tracker that can point a directional antenna toward the aircraft. Antennas for receiving video signals can be installed on the same tracker. 

The antenna trackers (such as Ardupilot's) however, use a protocol (Mavlink) that requires a special transmitter (basically requiring a link similar to Airport) 

My proposal is to create a version of the ELRS fw for the rx, which can receive telemetry transmitted by another rx, knowing its passphrase. By equipping this rx with a directive antenna we will have in a one-way link that allow to receive the telemetry and can provide the controller antenna tracker sw with the GPS coordinates of the aircraft and allow it to orient itself to get the best telemetry (and video signal).

![image](https://github.com/ExpressLRS/ExpressLRS/assets/11772150/c3e65fb3-bce7-4e57-986d-df2935ed55ea)

**Comment by sdellava** [comment]
Q&A: 

Q: why not use the radio to export telemetry to the antenna tracker controller?
A: because ERLS, with normal omnidirectional antennas, does not need a tracker antenna to ensure excellent link quality. The tracker antenna is used to point the video antennas, which instead with omnidirectional antennas have very limited link quality.

Q: why not also connect the radio to the directive antenna moved to the tracker?
A: because it is not essential, it would make the system more complex.

Q: why not to use Airport?
A: as already mentioned in the description, Airport is a very good solution but requires space on the aircraft and a base station possibly connected to a tracker. In contrast, the ELRS telemetry-based tacker is a very simple and independent device.

Q: Is there a tracker to which you could connect the receiver you envisioned?
A: Receiving the telementry data stream from the ELRS rx is very simple, so any tracker system can be upgraded to integrate this option.

**Reply by jstromsoe** [reply]
Why not just connect the ELRS receiver on the radio to the device pointing the antenna and allow either mavlink or telemetry of your choice to be streamed over the WiFi link?  Most ELRS receivers have WiFi now for OTA updating.  And lots of SBCs that have GPIO to control servos (RaspberryPi Zero W) have WiFi as well.

**Reply by sdellava** [reply]
Because it is simplier to use an FC. 

Anyway I've solved in a different way: https://github.com/sdellava/EDGETX-LUA-micro-antenna-tracker

Basically instead of using a separate "computer" I've used the the telemetry receiver (the EdgeTx) to excecte the required computation and drive the servos.

Anyway, I guess that a modular approach to the ELRS architecture could be usifull, so If one need just the telemetry, why not allow hit to go that way?

**Reply by jorgeneo560** [reply]
your solution is very good, but having another tx working on the same freq. could possible affects range, but if elrs is on 900mhz and the tx is 2.4ghz no problem, anyway it would be a nice addition to elrs to had a antenna tracker support, while more simple is the setup is more enjoyble to use and fly

**Comment by sdellava** [comment]
The second rx is reviewing only. Telemetry is disabled.
There is no possibility to produce interference.

Il gio 15 feb 2024, 15:46 jorgeneo560 ***@***.***> ha scritto:

**Comment by HGSAFD8162** [comment]
Hi, I've implemented the passive telemetry-sniffing receiver that was originally proposed in this discussion.
The project is called ExpressLRS Ghost RX: https://github.com/HGSAFD8162/Expresslrs-Ghost-RX
It is a modified ELRS 3.5.0 receiver firmware that:

Binds to the monitored TX to obtain the UID and track FHSS hopping
During telemetry response slots, stays in RX mode instead of transmitting — passively captures the real airborne RX's telemetry packets
Reassembles captured telemetry into CRSF frames and outputs them over UART
Requires zero changes to either the TX or the airborne RX firmware

The key property is that it never intentionally transmits, so there is no interference to the original link.
Currently tested on a BetaFPV Nano 2.4GHz RX (ESP8285 + SX1280), but the approach should be portable to other ELRS RX targets through the unified target / device configuration system.
Possible use cases beyond antenna tracking include standalone ground telemetry displays, flight data logging, and lost-model recovery with a directional antenna.
It's still at an early development stage with limited hardware validation. Feedback and testing from the community would be very welcome.

**Reply by HGSAFD8162** [reply]
I have already tested this on an existing ELRS link. Without interfering with the original communication, I can reliably receive telemetry from the aircraft — including GPS coordinates, battery voltage, flight mode, etc. — at distances up to 3 km, with the airborne RX transmitting telemetry at 100 mW. I am currently using it as a sidecar ground auxiliary device.
